store/storetest/link_metadata_store.go

```diff
@@ -136,10 +136,8 @@ func testLinkMetadataStoreGet(t *testing.T, ss store.Store) {
 		require.Nil(t, result.Err)
 
 		linkMetadata, err := ss.LinkMetadata().Get(metadata.URL, metadata.Timestamp)
-
 		require.Nil(t, err)
-		require.IsType(t, metadata, linkMetadata.Data)
-		assert.Equal(t, *metadata, *linkMetadata.Data.(*model.LinkMetadata))
+		assert.Equal(t, &model.PostImage{}, linkMetadata.Data)
```

Comment:
This is not the same comparison than before, the comparison should be something like:
`assert.Equal(t, *metadata, *linkMetadata)`

The previous one was using `result.Data` which was where the return value on success was previously, which have no relation at all with the `LinkMetadata.Data` field, which is another thing.

Reply:
I think I first totally ignored the fact that we were testing the value returned. I'm not sure if we should keep `require.IsType(t, metadata, linkMetadata` as `assert.Equal(t, *metadata, *linkMetadata` will blow if they are not of the same type.

Reply:
No, you don't need to keep that, that was before where we pass the result as a pointer that we need to cast, but that is not longer happening with the new approach, so the type is right because if not the compiler would complain.